### PR TITLE
Introducing Oppiabot configure file

### PR DESCRIPTION
**Pull request by anandwana001**
## Checklist

- [ ] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [ ] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

### CODE REVIEW

**Comment by anandwana001** [comment]
@vojtechjelinek @gp201 PTAL

File: .github/workflows/oppiabot.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# Contains jobs corresponding to Oppiabot.
+
+name: Oppiabot
+
+on:
+  pull_request_target:
+    branches:
+      - develop
```
Comment (gp201) [inline .github/workflows/oppiabot.yml:8]: On what branches should the CLA check be enabled?

**Review by gp201 — APPROVED** [review]
left a comment. LGTM

File: .github/workflows/oppiabot.yml (hunk 1 above)
Comment (anandwana001) [reply .github/workflows/oppiabot.yml:8]: As we need CLA (Oppiabot) for all the Pull request to take action, how should we mention the branch?
Also, will it work for fork branches as well?

Comment (gp201) [reply .github/workflows/oppiabot.yml:8]: When you mention "develop" here: https://github.com/oppia/oppia-android/blob/f8a5a936c8d9adf8f1ada6227362d223b3cb5604/.github/workflows/oppiabot.yml#L5-L8

It means that the CLA Check will run on any PR that has the develop branch as its base.
Do we need the check to run on all branches?
And could you rephrase "Also, will it work for fork branches as well?"?? I'm not able to understand?

Comment (anandwana001) [reply .github/workflows/oppiabot.yml:8]: This yml file is using secrets, which PRs from fork might not have access, so will the CLA works as we required?

Comment (gp201) [reply .github/workflows/oppiabot.yml:8]: Yes, the PR's from a fork will be able to access the secrets since it runs on the event. `pull_request_target`. ([Source](https://docs.github.com/en/actions/reference/events-that-trigger-workflows#pull_request_target)).

If you have doubts regarding the security please read the doc I had shared.

**Review by gp201 — CHANGES_REQUESTED** [review]
Adding myself as blocking so @vojtechjelinek can work with Sean to set the secret up once this is reviewed.

File: .github/workflows/oppiabot.yml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+jobs:
+  oppiabot:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: CLA Check
+        uses: oppia/oppiabot@1.2.0
+        env:
+          SHEETS_TOKEN: ${{ secrets.SHEETS_TOKEN }}
```
Comment (BenHenning) [inline .github/workflows/oppiabot.yml:18]: I don't think this is going to work. We can't use secrets like this for forks since they will be exposable to users.

**Comment by gp201** [comment]
@BenHenning PTAL

File: .github/workflows/oppiabot.yml [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+          SHEETS_CRED: ${{ secrets.SHEETS_CRED }}
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          SPREADSHEET_ID: ${{ secrets.SPREADSHEET_ID }}
```
Comment (gp201) [inline .github/workflows/oppiabot.yml]:
```suggestion
        uses: oppia/oppiabot@1.3.0
        with:
          repo-token: ${{secrets.GITHUB_TOKEN}}
        env:
          SHEETS_TOKEN: ${{ secrets.SHEETS_TOKEN }}
          SHEETS_CRED: ${{ secrets.SHEETS_CRED }}
          SPREADSHEET_ID: ${{ secrets.SPREADSHEET_ID }}
```

Oppiabot made a new release with some changes to CLA Check Github Action

**Comment by gp201** [comment]
@anandwana001 left a change

File: .github/workflows/oppiabot.yml (hunk 2 above)
Comment (BenHenning) [inline .github/workflows/oppiabot.yml:20]: Please follow up with me regarding which secrets need to be added & what their values are, otherwise this will fail when checked in.

**Review by BenHenning — APPROVED** [review]
Thanks @anandwana001. This LGTM given it's using pull_request_target. Please follow up with me regarding the secrets before checking this in.

Also, please follow up with me with examples for both fork & non-fork PRs to validate that the check is working (we should have an example for both contributor & non-contributor, fork & non-fork).

File: .github/workflows/oppiabot.yml (hunk 3 above)
Comment (anandwana001) [reply .github/workflows/oppiabot.yml]: Done

File: .github/workflows/oppiabot.yml (hunk 2 above)
Comment (anandwana001) [reply .github/workflows/oppiabot.yml:20]: @gp201 Could you follow up here?
> which secrets need to be added & what their values are, otherwise this will fail when checked in.

Comment (gp201) [reply .github/workflows/oppiabot.yml:20]: Secrets can be added at the organization level and the repository level. I'm not sure what approach is being taken currently. We have to consult @vojtechjelinek for the secrets.

Comment (vojtechjelinek) [reply .github/workflows/oppiabot.yml:20]: Since we share the CLA sheet between web and Android I think we can add this on org. level. So @seanlip should be able to add those.

Comment (BenHenning) [reply .github/workflows/oppiabot.yml:20]: We actually have a lot of repos under the org now. It would probably better to keep these repo level to restrict access. 

@seanlip do you have any thoughts on this?

Comment (gp201) [reply .github/workflows/oppiabot.yml:20]: @BenHenning Just an FYI, We have to give permission for the GitHub action to run on a repo in [this file](https://github.com/oppia/oppiabot/blob/65f1717d62f4f3358db40909e3c4228398ee14c1/constants.js#L61) in oppiabot.

Comment (BenHenning) [reply .github/workflows/oppiabot.yml:20]: Also, shouldn’t we be passing these values into the action rather than setting them as environment variables? While the env context is limited to this script, it seems a bit more secure to pass the secrets directly to the action like we are for the github token.

Comment (seanlip) [reply .github/workflows/oppiabot.yml:20]: > We actually have a lot of repos under the org now. It would probably better to keep these repo level to restrict access.
> @seanlip do you have any thoughts on this?

Would it be possible to explain more why having an org-level secret results in additional access issues -- who can access it and how would they do so? Do forks have access to org-based actions but not repo-based actions?

**Review by vinitamurthi — APPROVED** [review]

File: .github/workflows/oppiabot.yml (hunk 2 above)
Comment (seanlip) [reply .github/workflows/oppiabot.yml:20]: Hi -- I had a quick chat with @BenHenning on this and I think the core concern is about secrets for one repo in the org being leaked by another repo accidentally through careless handling.

I think this makes sense, we don't know what future repos we may create and what our policies on those would be. So let's go with repo-level for now.

Comment (gp201) [reply .github/workflows/oppiabot.yml:20]: > Also, shouldn’t we be passing these values into the action rather than setting them as environment variables? While the env context is limited to this script, it seems a bit more secure to pass the secrets directly to the action like we are for the github token.

@BenHenning 
I could add this as an issue on Github and make the change in a later PR so that it will change in the next release. I don't think it would make much of a difference, but it's always better to have it.

**Comment by seanlip** [comment]
Just a heads-up -- secrets have been added. How do we test that things are working correctly?

File: .github/workflows/oppiabot.yml (hunk 2 above)
Comment (BenHenning) [reply .github/workflows/oppiabot.yml:20]: @gp201 I'm fine with that. Can you please link to that issue here?

**Review by BenHenning — COMMENTED** [review]
Generally LGTM, but waiting for the open thread to get wrapped up before approving.

**Review by BenHenning — CHANGES_REQUESTED** [review]
Hmm, flipping back to 'request changes' per previous comment. I'm also interested in seeing this tested before merging it into develop.

**Comment by gp201** [comment]
Steps to test the GitHub action in the Oppia-Android repository.

1. Create a new branch in the Oppia-Andoird repo for testing the action
2. Change the branch `develop` to the new branch created. https://github.com/oppia/oppia-android/blob/c497bb3429c45705f016f6191d746dedae674d6b/.github/workflows/oppiabot.yml#L7-L8
3. Merge the PR to the **NEW** branch.
4. Make a PR to the **NEW** branch
5. The GitHub action should run.

This has already been performed in the Oppia repository.
PR that was used for testing: https://github.com/oppia/oppia/pull/13136
To test this please close and re-open the PR.

File: .github/workflows/oppiabot.yml (hunk 2 above)
Comment (gp201) [reply .github/workflows/oppiabot.yml:20]: @BenHenning The issue: https://github.com/oppia/oppiabot/issues/275

**Comment by gp201** [comment]
@BenHenning PTAL

**Comment by BenHenning** [comment]
Thanks for the instructions @gp201 they were really helpful! Per #3438 & #3439 it appears that Oppiabot is working correctly. Going ahead and approving this.

**Review by BenHenning — APPROVED** [review]
Thanks @anandwana001. LGTM.
